File: examples/rust/race_condition.rs
use std::thread;

fn main() {
    let words = vec!["Hello", "World", "From", "Rust"];
    words.push(" lol ");

    let threads = for w in words.clone() {
        let _ = thread::spawn(move ||{
            println!("{}", w);
        });
    };
    threads.join()
}

// Running
// cd examples/rust/
// rustc race_condition.rs && ./race_condition

File: examples/ruby/race_condition.rb
words = ["Hello", "world", "From", "Ruby"]

words.each_with_index.map do |w, i|
  Thread.new do
    words.push(" lol ")
    puts "#{i} #{words[i]}"
  end
end.each(&:join)


# Running
# cd example/ruby/
# ruby race_condition.rb
